app/forms/work_deposit_pathway.rb

```diff
@@ -33,6 +33,10 @@ def allows_curation_request?
     data_and_code? && !@resource.draft_curation_requested
   end
 
+  def allows_mint_doi_request?
+    data_and_code? && @resource.doi_blank? && DoiMintingStatus.new(@resource.work).blank?
```

Comment:
I think we should have some unit tests for this, since there's a handful of conditions.